### PR TITLE
Form errors

### PR DESCRIPTION
**Pull request by abutterworth**
- Each field now utilizes a form element and captures onSubmit events instead of onClick events from a save button.
- Errors on each field driven by props supplied to the ProfilePage

### CODE REVIEW

File: src/components/ProfilePage.jsx [hunk 1]
```diff
@@ -296,7 +295,17 @@ const mapStateToProps = (state) => {
...
+    errors: {
+      fullName: 'Your name must be at least two characters long.',
+      socialLinks: {
+        twitter: 'Please use a proper URL.',
+        facebook: 'Please use a proper URL.',
+        linkedin: 'Please use a proper URL.',
+      },
+      userLocation: 'Please choose another country.',
+      education: 'Please choose another school.',
+      bio: 'Please avoid XSS attempts.',
```
Comment (albemarle) [inline src/components/ProfilePage.jsx]: 😆

File: src/components/ProfilePage/SocialLinks.jsx [hunk 2]
```diff
@@ -197,7 +208,7 @@ function EditableListItem({
   onClickEmptyContent,
   name,
 }) {
-  const linkDisplay = url != null ?
+  const linkDisplay = (url != null && url !== '') ?
```
Comment (albemarle) [inline src/components/ProfilePage/SocialLinks.jsx]: _.isEmpty() ?

Comment (abutterworth) [reply src/components/ProfilePage/SocialLinks.jsx]: I think at this point could is just be `url ? something : somethingElse`
?

**Comment by davidjoy** [comment]
This PR has been subsumed by #40.  @abutterworth, I suggest closing it without merging. 👍

**Pull request closed without merging**